src/lint/rules/indentation.rs

```diff
@@ -1,11 +1,42 @@
 use std::convert::TryInto;
 
-use crate::{analysis::LocalDMLError, span::{Range, ZeroIndexed}};
+use crate::analysis::parsing::{expression::{CastContent, FunctionCallContent, ParenExpressionContent},
+                               lexer::TokenKind,
+                               statement::{self, CompoundContent, DoContent, ForContent, ForeachContent,
+                                           IfContent, SwitchCase, SwitchContent, WhileContent},
+                               structure::{MethodContent, ObjectStatementsContent},
+                               tree::TreeElementTokenIterator,
+                               types::{BitfieldsContent, LayoutContent, StructTypeContent}};
+use crate::span::{Range, ZeroIndexed};
+use crate::analysis::LocalDMLError;
+use crate::analysis::parsing::tree::{ZeroRange, Content, TreeElement};
 use serde::{Deserialize, Serialize};
 use super::Rule;
+use crate::lint::{LintCfg, DMLStyleError, RuleType};
 
 pub const MAX_LENGTH_DEFAULT: u32 = 80;
+pub const INDENTATION_LEVEL_DEFAULT: u32 = 4;
 
+fn default_indentation_spaces() -> u32 {
+    INDENTATION_LEVEL_DEFAULT
+}
+
+pub fn setup_indentation_size(cfg: &mut LintCfg) {
```

Comment:
This seems weird to me, does our rules config not support one setting being propagated to several rules? (as a consequence of us just passing the rule options sub-struct to the rule, I guess?) Overwriting config field from one rule to another seems incorrect to me. Perhaps define a set of 'global' settings shared by more than one rule and pass that in addition to local rules?

Reply:
We initially had the intention to make the original parameter a global variable, and have the options from the other rules have a reference to this global variable.
We did face some issues with the handling of lifetimes, and were not quite sure if this global variable would need to be handled as a static variable.
It would be nice to revisit this. Let us know if you have a suggestion or example on how to define the global value for this parameter.

Reply:
A truly global variable should _not_ be used. Rather we need a lintcfg variable not attached to any particular rule, that is then passed to all rules.

Reply:
Actually. Cant this go into aux_params, somehow?

Reply:
@JonatanWaern ping here, do you agree we can scope this to a follow up PR ?

Reply:
There is still some slight ugliness (if someone manually enters an indentation value for, for example, in3 in the cfg it will be ignored in favor of the in1 value. but if in1 for whatever reason is disabled then it would not be). (so, perhaps, what we need is a way to define cfg per-rule that is not settable from user cfg. but that would require a larger refactor I think?)

It feels to me like someone would have to intentionally try to do the above in order to get weird behavior though, so it's not of that much concern.  Agreed w/ moving this to later, making a mental note to create an issue for it after merging.

Reply:
https://github.com/intel/dml-language-server/issues/100

I have documented this here. Can you tag it as an enhancement (or any other label you would use for this)